tests/tokenize_cmd_test.c

```diff
@@ -0,0 +1,118 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*                                                        :::      ::::::::   */
+/*   tokenize_cmd_test.c                                :+:      :+:    :+:   */
+/*                                                    +:+ +:+         +:+     */
+/*   By: taretiuk <taretiuk@student.42.fr>          +#+  +:+       +#+        */
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2024/11/04 06:11:45 by taretiuk          #+#    #+#             */
+/*   Updated: 2024/11/04 06:27:29 by taretiuk         ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include "tests_internal.h"
+#include "../sources/tokenizing/input_processing.h"
+
+#define NUM_TEST_CASES 11
+#define MAX_ARGS 16
+
+typedef struct s_string
+{
+	char	*str;
+}	t_string;
+
+typedef struct s_string_array
+{
+	t_string	*strs;
+	int			error;
+	size_t		count;
+}	t_strings;
+
+t_strings	create_string_array()
+{
+	t_strings	str_array;
+	str_array.count = 11;
+	str_array.error = 0;
+	str_array.strs = ft_calloc_if(sizeof(t_string) * str_array.count, 1);
+	assert(str_array.strs != NULL);
+	for (int i = 0; i < (int)str_array.count; i++)
+	{
+		str_array.strs[i].str = NULL;
+	}
+	str_array.strs[0].str = strdup("grep \"error\" log.txt || echo \"No errors    found\"");
+	str_array.strs[1].str = strdup("cat ( file1.txt file2.txt ) | grep \"keyword\"");
+	str_array.strs[2].str = strdup("cat < input.txt && echo | grep \"Done\"");
+	str_array.strs[3].str = strdup("gcc main.c -o program >> build.log");
+	str_array.strs[4].str = strdup("find . -name \"*.txt\" | xargs rm -rf || fer && def & echo \"Hello\"");
+	str_array.strs[5].str = strdup("echo \"Compiling...\" && gcc main.c -o program && ./program");
+	str_array.strs[6].str = strdup("mkdir new_dir && cd new_dir && touch file.txt");
+	str_array.strs[7].str = strdup("mv new_dir old_dir");
+	str_array.strs[8].str = strdup("&&");
+	str_array.strs[9].str = strdup(")(())(");
+	str_array.strs[10].str = strdup("");
+	return (str_array);
+}
+
+static void	free_tokens(char **tokens)
+{
+	if (!tokens)
+		return;
+	for (int i = 0; tokens[i] != NULL; i++)
+	{
+		free(tokens[i]);
+		tokens[i] = NULL;
+	}
+	free(tokens);
+}
+
+static void	free_string_array(t_string *strs, size_t count) 
+{
+	if (!strs)
+		return;
+
+	for (size_t i = 0; i < count; i++) 
+	{
+		free(strs[i].str);
+	}
+	free(strs);
+}
+
+void	tokenize_cmd_test()
+{
+	t_strings	str_arr = create_string_array();
+	char		**tokens;
+
+	assert(str_arr.error == 0 && "Failed to create string array.");
+	char	*t[NUM_TEST_CASES][MAX_ARGS] =
+	{
+		{"grep", "\"error\"", "log.txt", "||", "echo", "\"No errors    found\"", NULL},
```

Comment:
I think we [don't have to store](https://github.com/ioannnikulin/minishell/blob/dev/sources/input_to_text_tree_mocks/mock_2.c#L18) block delimiters (double quotes) in the token. Do you want to omit them now in split, or omit them later when creating a tree, or make omitting a changeable option?

Reply:
As long as I get the result I expect (as shown in the mocks), I'm fine with any decision on your side

Reply:
Then you can mark as resolved